service/api/cadence/client.go

```diff
@@ -143,6 +149,21 @@ func mapToIwfSearchAttributes(searchAttributes *shared.SearchAttributes) (map[st
 	return result, nil
 }
 
+func mapToTemporalWorkflowIdReusePolicy(workflowIdReusePolicy string) (client.WorkflowIDReusePolicy, error) {
+	switch workflowIdReusePolicy {
+	case service.WorkflowIDReusePolicyAllowDuplicate:
+		return client.WorkflowIDReusePolicyAllowDuplicate, nil
+	case service.WorkflowIDReusePolicyAllowDuplicateFailedOnly:
+		return client.WorkflowIDReusePolicyAllowDuplicateFailedOnly, nil
+	case service.WorkflowIDReusePolicyRejectDuplicate:
+		return client.WorkflowIDReusePolicyRejectDuplicate, nil
+	case service.WorkflowIDReusePolicyTerminateIfRunning:
+		return client.WorkflowIDReusePolicyTerminateIfRunning, nil
+	default:
+		return client.WorkflowIDReusePolicyAllowDuplicate, fmt.Errorf("unsupported workflow id reuse policy %s", workflowIdReusePolicy)
```

Comment:
cadence doesn't have a UNSPECIFIED enum

Reply:
yeah returnin gerror should be fine

Reply:
we probably should return nil for the policy (need to change the signature to use pointer)